File: app/src/main/java/com/mc10inc/codetest/base/BaseActivity.java
package com.mc10inc.codetest.base;

import android.support.v7.app.AppCompatActivity;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * TODO
 *
 * @author Adam Stroud &#60;<a href="mailto:adam.stroud@gmail.com">adam.stroud@gmail.com</a>&#62;
 */
public abstract class BaseActivity extends AppCompatActivity {
    private final CompositeDisposable compositeDisposable = new CompositeDisposable();

    protected Disposable manage(Disposable disposable) {
        compositeDisposable.add(disposable);
        return disposable;
    }

    @Override
    protected void onDestroy() {
        compositeDisposable.clear();
        super.onDestroy();
    }
}

File: app/src/main/kotlin/com/mc10inc/codetest/base/BaseActivity.kt
package com.mc10inc.codetest.base

import android.support.v7.app.AppCompatActivity
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable
import io.reactivex.rxkotlin.addTo

/**
 * TODO
 *
 * @author Adam Stroud &#60;[adam.stroud@gmail.com](mailto:adam.stroud@gmail.com)&#62;
 */
abstract class BaseActivity : AppCompatActivity() {
    private val compositeDisposable = CompositeDisposable()

    override fun onDestroy() {
        clearDisposables()
        super.onDestroy()
    }

    protected fun manage(disposable: Disposable?) = disposable?.addTo(compositeDisposable)

    protected fun clearDisposables() {
        compositeDisposable.clear()
    }
}